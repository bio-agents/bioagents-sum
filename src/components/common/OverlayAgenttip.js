import React from 'react'
import { OverlayTrigger, Agenttip } from 'react-bootstrap'

const OverlayAgenttip = ({ id, agenttipText, children, delayShow = 300, delayHide = 150, placement = 'top' }) => {
  const agenttip = <Agenttip id={id}>{agenttipText}</Agenttip>

  return (
    <OverlayTrigger
      overlay={agenttip}
      placement={placement}
      delayShow={delayShow}
      delayHide={delayHide}
    >
      {children}
    </OverlayTrigger>
  )
}

export default OverlayAgenttip
