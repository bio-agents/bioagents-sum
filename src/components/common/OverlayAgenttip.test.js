import React from 'react'
import renderer from 'react-test-renderer'
import OverlayAgenttip from './OverlayAgenttip'

it('Renders OverlayAgenttip correctly', () => {
  const tree = renderer
    .create(
      <OverlayAgenttip id='overlay_snapshot' agenttipText='This is overlay snapshot'>
        <span>{'OverlayAgenttip child'}</span>
      </OverlayAgenttip>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
