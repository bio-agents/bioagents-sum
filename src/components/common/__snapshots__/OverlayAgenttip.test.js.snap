exports[`test Renders OverlayAgenttip correctly 1`] = `
<span
  onBlur={[Function]}
  onClick={null}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}>
  OverlayAgenttip child
</span>
`;
